### DISCUSSION TITLE
How high does someone's IQ have to be to understand low-level programming like assembly or binary code and programming in general?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#141723] [Programming Help]
### Body

How high does someone's IQ have to be to understand low-level programming like assembly or binary code and programming in general?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dd3vahmad** [comment]
@WorldsDumbestCoder 😂, understanding low-level programming like assembly or binary code doesn't require an exceptionally high IQ. It's more about developing specific skills, logical thinking, and patience over time. Here's why:

1. **Learning Mindset**: Low-level programming involves understanding how computers process instructions at a basic level. This can be learned with the right resources and practice, much like learning a new language. You don’t need an exceptionally high IQ, but you do need to be methodical, detail-oriented, and persistent.

2. **Problem-Solving Skills**: Problem-solving is key to programming in general, and those who have a natural inclination toward logical thinking might find learning low-level languages easier. IQ might help with abstract thinking, but structured practice can improve problem-solving.

3. **Focus and Patience**: Assembly code, for example, requires careful attention to detail and patience due to its complexity and verbosity. A person needs to break down tasks into smaller steps, which is a skill that can be honed regardless of IQ.

4. **Resources and Support**: Access to good learning resources, mentors, or communities is crucial for mastering these concepts. Anyone with determination can reach a deep understanding with consistent effort.

In conclusion, while a high IQ might give someone a slight advantage in grasping abstract concepts, it’s not a requirement. Persistence, practice, and a systematic approach are much more important when learning low-level programming.

**Reply by ghost** [reply]
```php
<?php
$sarcasm = true;
if($sarcasm == true){
     echo "Are you sure? I suck at assembly.\n";
}else{
   echo "Thanks!\n";
}
?>
```
